asm/f64/sdot.go

```diff
@@ -4,18 +4,18 @@
 // Use of this source code is governed by a BSD-style
 // license that can be found in the LICENSE file.
 
-package asm
+package f64
 
-func SdotUnitary(x, y []float32) (sum float32) {
+func SdotUnitary(x, y []float32) (sum float64) {
 	for i, v := range x {
-		sum += y[i] * v
+		sum += float64(y[i]) * float64(v)
 	}
 	return
 }
 
-func SdotInc(x, y []float32, n, incX, incY, ix, iy uintptr) (sum float32) {
+func SdotInc(x, y []float32, n, incX, incY, ix, iy uintptr) (sum float64) {
```

Comment:
I did move sdot.go, dsdot.go, and ddot.go incorrectly.  After renaming dsdot.go to sdot.go, I mixed up the two sdot.go files.  It ended up correct, but the two files got swapped en route.  I couldn't get that fixed in rebase.